validator/api/src/main/java/tech/pegasys/teku/validator/api/UpdatableGraffitiProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class UpdatableGraffitiProvider implements GraffitiProvider {
+  private final Supplier<Optional<Bytes32>> storageProvider;
+  private final GraffitiProvider defaultProvider;
+
+  public UpdatableGraffitiProvider(
+      final Supplier<Optional<Bytes32>> storageProvider, final GraffitiProvider defaultProvider) {
+    this.storageProvider = storageProvider;
+    this.defaultProvider = defaultProvider;
+  }
+
+  @Override
+  public Optional<Bytes32> get() {
+    return storageProvider.get().or(defaultProvider::get);
```

Comment:
I think it would be safer to encapsulate the call to the `storageProvider` function into a try-catch block to avoid any surprises. Or do something similar to what is done in the `FileBackedGraffitiProvider` (that has a try-catch Exception block to handle ANY errors). No matter how we do it, we need to ensure this won't fail during block production.

The logic should be: no matter what explodes in the storageProvider, we want to ignore it and delegate to the defaultProvider.

The following test should pass:
```
  @Test
  @SuppressWarnings("unchecked")
  public void shouldDelegateToDefaultProviderWhenStorageProviderFails() {
    final Supplier<Optional<Bytes32>> fn = mock(Supplier.class);
    when(fn.get()).thenThrow(new RuntimeException("Error"));

    provider =
        new UpdatableGraffitiProvider(fn, () -> Optional.of(defaultGraffiti));
    assertThat(provider.get()).hasValue(defaultGraffiti);
  }
```